main.go

```diff
@@ -197,6 +200,141 @@ type MainController struct {
 	db *sqlx.DB
 }
 
+func (c MainController) authAttendanceMiddleware(h http.Handler) http.Handler {
```

Comment:
These three functions look identical except the allowedRoles slices. I'd refactor them to use a common `authRolesMiddleware(allowedRoles []string) func(http.Handler) http.Handler` method.

Reply:
I would prefer to use a common method, but couldn't figure out how to write a middleware that accepts a parameter. Maybe it's my lack of experience with Go. Any help on that would be much appreciated.

Reply:
Probably easiest:

    func (c MainController) authAttendeeMiddleware(h http.Handler) http.Handler {
        return c.authRoleMiddleware(c, []string{...})
    }

    func (c MainController) authRoleMiddleware(h http.Handler, allowedRoles []string) http.Handler {
        // Same code as below, except using the allowedRoles parameter instead of defining it as a local variable.
    }